Library/Homebrew/formula_installer.rb

```diff
@@ -351,7 +351,7 @@ def install
       message = <<~EOS
         #{formula.name} #{formula.linked_version} is already installed
       EOS
-      if formula.outdated? && !formula.head?
+      if Homebrew::EnvConfig.no_install_upgrade? && formula.outdated? && !formula.head?
```

Comment:
Sorry to be a pain: I wonder if this logic could/should be in `upgrade.rb` and called from `formula_installer` too? Feels weird to have this messaging scattered across files.

Reply:
I've moved all of that code into `install.rb` as it all only used with `brew install`.